Task-2/Task-2

```diff
@@ -0,0 +1,164 @@
+create table dim_customers(
+	customer_sk serial primary key ,
+	customer_id int not null,
+	name varchar(50),
+	email varchar(60),
+	effective_date date not null,
+	end_date date,
+	is_active char(1) check (is_active in ('Y','N'))
+);
+
+create table dim_products(
+	product_sk serial primary key ,
+	product_id int not null,
+	name varchar(50),
+	category varchar(50)
+);
+
+create table dim_date(
+	date_sk date primary key,
+	full_date date not null,
+	year int,
+	month int ,
+	day_of_week varchar(10)
+	
+);
+
+create table fact_orders(
+	order_id serial primary key,
+	order_date_sk date,
+	customer_sk int,
+	product_sk int,
+	quantity int,
+	price numeric(10,2),
+	total_amount numeric(10,2),
+	foreign key (order_date_sk) references dim_date(date_sk),
+	foreign key (customer_sk) references dim_customers(customer_sk),
+	foreign key (product_sk) references dim_products(product_sk)
+);
+
+
+insert into dim_customers 
+values
+(101,'Hamza','hamzakhan101@hotmail.com',date '9-3-2025',null,'Y'),
+(102, 'Ali', 'ali.khan@gmail.com', date '2025-03-08', NULL, 'Y'),
+(103, 'Sana', 'sana.bibi@yahoo.com', date '2025-03-05',  '2025-03-10', 'N'),
+(104, 'Usman', 'usman123@gmail.com', date '2025-03-01', NULL, 'Y'),
+(105, 'Ayesha', 'ayesha.malik@hotmail.com', date '2025-02-25', DATE '2025-03-05', 'N');
+
+select * from dim_customers ;
+
+insert into dim_products (product_id, name, category)
+values
+(1, 'Laptop', 'Electronics'),
+(2, 'Mobile', 'Electronics'),
+(3, 'Shirt', 'Clothing'),
+(4, 'Jeans', 'Clothing'),
+(5, 'Microwave', 'Home Appliances');
+
+select * from dim_products;
+
+insert into dim_date (date_sk, full_date, year, month, day_of_week)
+values
+(DATE '2025-03-01', DATE '2025-03-01', 2025, 3, 'Saturday'),
+(DATE '2025-03-02', DATE '2025-03-02', 2025, 3, 'Sunday'),
+(DATE '2025-03-03', DATE '2025-03-03', 2025, 3, 'Monday'),
+(DATE '2025-03-04', DATE '2025-03-04', 2025, 3, 'Tuesday'),
+(DATE '2025-03-05', DATE '2025-03-05', 2025, 3, 'Wednesday');
+
+
+select * from dim_date;
+
+insert into fact_orders
+(order_date_sk, customer_sk, product_sk, quantity, price, total_amount)
+values
+(DATE '2025-03-01', 1, 1, 1, 800, 800),
+(DATE '2025-03-01', 2, 2, 2, 500, 1000),
+(DATE '2025-03-02', 3, 3, 3, 30, 90),
+(DATE '2025-03-02', 4, 4, 2, 40, 80),
+(DATE '2025-03-03', 5, 5, 1, 150, 150),
+(DATE '2025-03-03', 1, 2, 1, 500, 500),
+(DATE '2025-03-04', 2, 3, 2, 30, 60),
+(DATE '2025-03-04', 3, 4, 1, 40, 40),
+(DATE '2025-03-05', 4, 5, 2, 150, 300),
+(DATE '2025-03-05', 5, 1, 1, 800, 800);
+
+select * from fact_orders;
+
+
+update dim_customers
+set end_date = date '2025-03-15', is_active='N'
```

Comment:
In SCD-2 you cannot set only date as start or end dates, it lose precision, you need exact timestamp.